src/main/java/pl/firstService/employeeApi/controller/CenterApiController.java

```diff
@@ -0,0 +1,102 @@
+import pl.firstService.employeeApi.model.Center;
+import pl.firstService.employeeApi.service.CenterService;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/center")
+public class CenterApiController {
+    private final CenterDtoMapper centerDtoMapper = new CenterDtoMapper();
+    private final CenterService centerService;
+
+    //TODO dopisać validatory, actuator, testy jednostkowe, przerobić na zwrot Response Entity, post może zwracać tylko id ale link do geta
+    //Hateoas dać wszędzie
+
+    @GetMapping()
+    @Operation(summary = "Get list of centers")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Ok", content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = List.class))),
+            @ApiResponse(responseCode = "500", description = "Server error", content = @Content)
+    })
+    public List<CenterDto> getCenters() {
+        List<Center> centers = centerService.getCenters();
+        return centers.stream()
+                .map(centerDtoMapper::convertToDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Get center by id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Ok", content = @Content(mediaType = "application/json",
+            schema = @Schema(implementation = CenterDto.class))),
+            @ApiResponse(responseCode = "404", description = "Center not found", content = @Content),
+            @ApiResponse(responseCode = "500",description = "Server error", content = @Content)
+    })
+    public ResponseEntity getCenterById(@PathVariable Long id) {
+        Optional<Center> entity = centerService.getCenterById(id);
+        if(!entity.isPresent()){
+            return new ResponseEntity(String.format("Center with id %s not found",id),HttpStatus.NOT_FOUND);
+        }
+        CenterDto centerDto = centerDtoMapper.convertToDto(entity.get());
+        return ResponseEntity.ok(centerDto);
+    }
+
+
+
+    @PostMapping()
+    @Operation(summary = "Create a center")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Center created", content = @Content(mediaType = "application/json",
+            schema = @Schema(implementation = CenterCreateDto.class))),
+            @ApiResponse(responseCode = "400", description = "Bad request", content = @Content(mediaType = "application/json",
+            schema = @Schema(implementation = CenterValidationErrorDto.class))),
+            @ApiResponse(responseCode = "500",description = "Server error", content = @Content)})
+    public ResponseEntity<CenterDto> createCenter(@RequestBody CenterCreateDto centerDto) {
```

Comment:
Czy na DTO miałybyć jakieś walidację? 
Jeśli tak to poczytaj o 
https://www.baeldung.com/javax-validation-method-constraints
https://www.baeldung.com/spring-valid-vs-validated